In [9]:
import numpy as np
import pandas as pd
import json
import datetime
import nltk
from nltk.corpus import stopwords
from operator import itemgetter 
import matplotlib.pyplot as plt

In [5]:
with open('submission/data.json') as file:
    data = file.read()
    new_data = data.replace('}{', '},{')
    json_data = json.loads(f'[{new_data}]')
    
submission = pd.DataFrame.from_records(json_data)

submission_timed = submission
submission_timed['time'] = pd.to_datetime(submission['time'])

In [6]:
gme_raw = pd.read_csv('GME.csv')

In [20]:
gme_timed = gme_raw 
gme_timed['Date'] = pd.to_datetime(gme_timed['Date'])

In [21]:
gme_date = gme_raw['Date'].to_list()

In [15]:
gme_close = gme_raw['Adj Close'].to_list()

In [10]:
submission_gme = submission_timed[submission_timed['time'] >= datetime.date(2020,8,14)]

/Users/xiaoyu/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: Comparing Series of datetimes with 'datetime.date'.  Currently, the
'datetime.date' is coerced to a datetime. In the future pandas will
not coerce, and a TypeError will be raised. To retain the current
behavior, convert the 'datetime.date' to a datetime with
'pd.Timestamp'.
  """Entry point for launching an IPython kernel.


In [28]:
date2submission = {}
date2count = {}

for date in gme_date:
    date2submission[date] = submission_gme[submission_gme['time'] == date]
    date2count[date] = date2submission[date]['title'].str.lower().str.count('gme').sum()

In [30]:
gme_count = [x for x in date2count.values()]

In [37]:
from scipy.stats import pearsonr
corr, _ = pearsonr(gme_count[:-1],gme_close[1:])
print(corr)

0.7703570005900581


The pearson correlation coefficient between the number of occurrences of 'gme' in title and the closing price next day is 0.77, which suggests that they are closely correlated. This shows the potential of building a prediction model with features generated from r/wallstreetbets